### ISSUE TITLE
Results of the paper vgg 16 Optical Flow

### ISSUE THREAD

**Issue by Chuseuiti** [yjxiong/caffe#45]
After running **train_action_recognition_flow.sh** with the optical flow data and the pre-trained model of the website, I was not able to achieve the 86%  for vgg_16 of the paper(Towards Good Practices for Very Deep Two-Stream ConvNets). I only achieved 68%, because of that I would like to know if vgg_16_flow_train_val_fast.prototxt is the correct prototxt of the paper.

**Comment by wanglimin** [comment]
Hi, could you provide more details about training such as number of GPUs, batch_size. BTW, did you extract optical flow by yourself or use the flow provided by us?

**Comment by Chuseuiti** [comment]
I used the _optical flow_ provided by you at mmlab.siat.ac.cn/very_deep_two_stream_model/ucf101_flow_img_tvl1_gpu.zip.

_Batch size: _
 video_data_param {
    source: "/examples/action_recognition/dataset_file_examples/train_flow_split1.txt"
   *\* batch_size: 22**
    new_length: 10
    num_segments: 1
    modality: FLOW
  }

  video_data_param {
    source: "/examples/action_recognition/dataset_file_examples/val_flow_split1.txt"
   *\* batch_size: 5**
    new_length: 10
    num_segments: 1
    modality: FLOW
  }

_One GPU_: NVIDIA Corporation Device 17c2

**Comment by yjxiong** [comment]
As noted in the read me, the effective batch size is `num_gpu * iter_size *
batch_size`. The setting you use was setup to run with _4_ GPUs. Since you
are using 1 GPU instead, please try compensate the decrease of effective
batch size by increasing either `iter_size` or `batch_size`. Otherwise you
will obtain inferior results.

On Sun, Feb 14, 2016 at 7:57 AM, Chuseuiti notifications@github.com wrote:

> I used the _optical flow_ provided by you at
> mmlab.siat.ac.cn/very_deep_two_stream_model/ucf101_flow_img_tvl1_gpu.zip.
> 
> _Batch size: _
> video_data_param {
> source:
> "/examples/action_recognition/dataset_file_examples/train_flow_split1.txt"
> *\* batch_size: 22**
> new_length: 10
> num_segments: 1
> modality: FLOW
> }
> 
> video_data_param {
> source:
> "/examples/action_recognition/dataset_file_examples/val_flow_split1.txt"
> *\* batch_size: 5**
> new_length: 10
> num_segments: 1
> modality: FLOW
> }
> 
> _One GPU_: NVIDIA Corporation Device
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/yjxiong/caffe/issues/45#issuecomment-183773559.

## 

熊元骏
Xiong, Yuanjun

Department of Information Engineering,
The Chinese University of Hong Kong.

E-mail: bitxiong@gmail.com
Mobile: +852 54282253
Mainland China Mobile: +86 147 1430 8814
Address: SHB703, The Chinese University of Hong Kong.
              Shatin, N.T, Hong Kong.

**Comment by Chuseuiti** [comment]
Taking that into account in the example the batch size shouldnt be 12 instead of 22, as there are 4 gpu and iter. size is 3 in the proto solver? and why is 5 in the test?

**Comment by Chuseuiti** [comment]
And why is 5 in the test ?

**Comment by yjxiong** [comment]
The effective training batch size is 22 \* 4 \* 3 = 264. Please make sure you are using a similar number. The testing batch size does not use the `iter_size` param, while the number of GPUs matters.

**Comment by yjxiong** [comment]
closing this. Please feel free to reopen if you still have problem reproducing the results.

**Comment by Chuseuiti** [comment]
After changing the batch size to:
22 \* 1 \* 12=264
and 
30 \* 1 \* 9= 270 

My best results have been:
loss accuracy
1.19 71%
1.18 74%
1.138 74.7%
1.06 75.5%
1.12 71.6%
1.17 73.1%
and
loss accuracy
1.072 74.9%
1.02 75.4%

But Im not near the 86% of the paper for optical flow, could you give me some tips on this please?

**Comment by yjxiong** [comment]
After running the CNN training. Please use the scripts provided by us to obtain the video based prediction results.
